packages/plans-grid-next/src/hooks/data-store/use-plan-for-spotlight.ts

```diff
@@ -0,0 +1,31 @@
+import { getPlanClass } from '@automattic/calypso-products';
+import { useMemo } from 'react';
+import { GridPlan } from '../../types';
+
+const SPOTLIGHT_ENABLED_INTENTS = [ 'plans-default-wpcom' ];
+
+interface Params {
+	gridPlansForFeaturesGrid: GridPlan[];
+	intent: string | undefined;
+	isSpotlightOnCurrentPlan?: boolean;
+	sitePlanSlug?: string | null;
+}
+
+const usePlanForSpotlight = ( {
+	gridPlansForFeaturesGrid,
+	intent,
+	isSpotlightOnCurrentPlan,
+	sitePlanSlug,
+}: Params ): GridPlan | undefined => {
+	return useMemo( () => {
+		const isIntentSpotlightEnabled = intent ? SPOTLIGHT_ENABLED_INTENTS.includes( intent ) : false;
+
+		return sitePlanSlug && isSpotlightOnCurrentPlan && isIntentSpotlightEnabled
+			? gridPlansForFeaturesGrid.find(
+					( { planSlug } ) => getPlanClass( planSlug ) === getPlanClass( sitePlanSlug )
```

Comment:
I noticed we are comparing CSS classes with these `getPlanClass()` calls, which doesn't seem ideal. However I'm unsure about why this approach was taken, perhaps there was a good reason I'm missing 🤔 

Should we instead just compare the slugs? 

`( { planSlug } ) => planSlug === sitePlanSlug`

Reply:
Ok thanks for the context @ddc22